src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -292,373 +58,392 @@ def test_dynbuiltfactory_str():
-                "      !\n"
+            output = (
+                "      f3_proxy = f3%get_proxy()\n"
                 "      f1_proxy = f1%get_proxy()\n"
+                "      f2_proxy = f2%get_proxy()\n"
                 "      !\n"
                 "      ! Initialise number of layers\n"
                 "      !\n"
-                "      nlayers = f1_proxy%vspace%get_nlayers()\n"
+                "      nlayers = f3_proxy%vspace%get_nlayers()\n"
                 "      !\n"
                 "      ! Initialise sizes and allocate any basis arrays for "
-                "any_space_1_f1\n"
+                "any_space_1_f3\n"
                 "      !\n"
-                "      ndf_any_space_1_f1 = f1_proxy%vspace%get_ndf()\n"
-                "      undf_any_space_1_f1 = f1_proxy%vspace%get_undf()\n"
+                "      ndf_any_space_1_f3 = f3_proxy%vspace%get_ndf()\n"
+                "      undf_any_space_1_f3 = f3_proxy%vspace%get_undf()\n"
                 "      !\n"
                 "      ! Call our kernels\n"
                 "      !\n"
-                "      DO df=1,undf_any_space_1_f1\n"
-                "        f1_proxy%data(df) = 0.0\n"
-                "      END DO \n"
-                "      !\n"
-                "    END SUBROUTINE invoke_0\n")
-            print output_seq
-            assert output_seq in code
-
-        if distmem:
-            mesh_code_present(code)
-            output_dm_2 = (
+                "      DO df=1,undf_any_space_1_f3\n"
+                "        f3_proxy%data(df) = f1_proxy%data(df) + "
+                "f2_proxy%data(df)\n"
+                "      END DO")
+            assert output in code
+        if distmem:
+            mesh_code_present("f3", code)
+            output_dm_2 = (
                 "      !\n"
                 "      ! Call kernels and communication routines\n"
                 "      !\n"
-                "      DO df=1,f1_proxy%vspace%get_last_dof_owned()\n"
-                "        f1_proxy%data(df) = 0.0\n"
+                "      DO df=1,f3_proxy%vspace%get_last_dof_owned()\n"
+                "        f3_proxy%data(df) = f1_proxy%data(df) + "
+                "f2_proxy%data(df)\n"
                 "      END DO \n"
                 "      !\n"
                 "      ! Set halos dirty for fields modified in the "
                 "above loop\n"
                 "      !\n"
-                "      CALL f1_proxy%set_dirty()\n"
+                "      CALL f3_proxy%set_dirty()\n"
                 "      !\n")
             print output_dm_2
             assert output_dm_2 in code
 
 
-def test_builtin_set_by_ref():
-    ''' Tests that we generate correct code for a builtin
-    set operation with a scalar passed by reference '''
-    _, invoke_info = parse(
-        os.path.join(BASE_PATH,
-                     "15.0.1_single_builtin_set_by_ref.f90"),
-        api="dynamo0.3")
+def test_inc_X_plus_Y_str():    # X = X + Y
```

Comment:
Again, please merge this test in with `test_inc_X_plus_Y`

Reply:
Merged as requested above and updated description.